### ISSUE TITLE
Issue 007: channel tiling 有効時に tile probability map を merge してから GPT する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#356]
# Issue 007: channel tiling 有効時に tile probability map を merge してから GPT する

## 背景

coarse global-anchor inference は channel tiling を持っている。現在の argmax pipeline は tile ごとに restored pick を作り、overlap weight で full-trace pick を merge している。しかし GPT では、tile ごとに path tracing すると tile-local path になり、global path ではなくなる。

GPT enabled の正しい順序は以下。

```text
for each channel tile:
    logits -> softmax probability
    anchor row probability を gather coordinate に蓄積

all tiles merged:
    merged probability -> log score
    segment-aware 2D GPT
    restore/interpolate to all traces in gather
```

Issue 006 では安全のため複数 tile を reject した。この issue でその制限を外す。

## 目的

複数 channel tile がある gather でも、tile probability map を merge した後に 1 回だけ GPT を行うようにする。

## 非目的

この issue では以下は行わない。

- 3D GPT
- raw waveform GPT
- physics/fine の変更
- QC overlay の完成。必要最小限の QC 無効化/維持は可。詳細 QC は Issue 008。

## 変更対象

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/infer.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/trace_anchor.py  # 必要なら helper export
packages/seisai-engine/tests/test_fbpick_coarse.py
packages/seisai-engine/tests/test_fbpick_trace_anchor.py  # 必要なら
```

必要に応じて helper module を追加する。

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/gpt_score_merge.py
packages/seisai-engine/tests/test_fbpick_coarse_gpt_score_merge.py
```

### 1. gather-level accumulator

新規 class を追加する。

```python
@dataclass
class CoarseGPTGatherScoreAccumulator:
    file_idx: int
    key_name: str
    primary_value: int
    gather_n_traces: int
    raw_time_len: int
    coarse_time_len: int
    dt_sec: float
    prob_weighted_sum: np.ndarray      # shape (gather_n_traces, coarse_time_len)
    weight_sum: np.ndarray             # shape (gather_n_traces,)
    raw_trace_indices: np.ndarray      # shape (gather_n_traces,), initialized -1
    offsets_m: np.ndarray              # shape (gather_n_traces,), initialized nan
    tile_seen: np.ndarray              # shape (tile_count,), bool
    tile_count: int
```

Keyed by:

```python
(file_idx, key_name, primary_value)
```

`primary_unique` string can be included for error messages, but primary key should be deterministic.

### 2. per-tile accumulation

For each tile output:

```python
logits_hw -> log_softmax_time -> prob_hw
```

For each valid anchor row:

```python
local_pos = anchor_source_pos[row]
gather_pos = channel_tile_start_pos + local_pos
anchor_weight = tile_weights[local_pos]
prob_weighted_sum[gather_pos, :] += prob_hw[row, :] * anchor_weight
weight_sum[gather_pos] += anchor_weight
```

Also fill:

```python
raw_trace_indices[channel_tile_start_pos:channel_tile_stop_pos] = meta['raw_trace_indices']
offsets_m[channel_tile_start_pos:channel_tile_stop_pos] = meta['offsets_m']
tile_seen[channel_tile_id] = True
```

Overlap validation:

- overlap span で既に raw_trace_indices が入っている場合、同じ値であること。
- offsets も finite で、既存値と close であること。
- `channel_tile_count`, `gather_n_traces`, `raw_time_len`, `coarse_time_len`, `dt_sec` が accumulator と一致すること。

### 3. flush condition

`tile_seen.all()` になったら gather を process する。

flush 時:

- `raw_trace_indices` が全て埋まっていること。
- `offsets_m` が全て finite。
- `weight_sum > 0` の rows を anchor rows とする。
- anchor positions are `np.flatnonzero(weight_sum > 0)`。
- `prob_anchors = prob_weighted_sum[positions] / weight_sum[positions, None]`。
- `prob_anchors` を row ごとに再正規化する。

再正規化:

```python
row_sum = prob_anchors.sum(axis=-1, keepdims=True)
prob_anchors = prob_anchors / row_sum
score_anchors = np.log(np.clip(prob_anchors, eps, 1.0)).astype(np.float32)
```

### 4. full-gather segment topology

tile-local `meta['segments']` は使わない。flush 時に full gather offsets から再計算する。

```python
segments = split_trace_segments_by_offset_gap(
    offsets_m,
    gap_ratio=typed.trace_anchor.gap_ratio,
    min_gap_m=typed.trace_anchor.min_gap_m,
)
```

各 anchor position に component id を割り当てる。

```python
anchor_segment_id = np.full(anchor_positions.shape, -1, dtype=np.int64)
for segment in segments:
    mask = (anchor_positions >= segment.start_pos) & (anchor_positions < segment.stop_pos)
    anchor_segment_id[mask] = segment.segment_id
```

### 5. GPT / argmax after merge

argmax baseline:

```python
anchor_argmax_j = prob_anchors.argmax(axis=-1)
anchor_argmax_pmax = prob_anchors.max(axis=-1)
```

GPT:

```python
ScoreVolume(score=score_anchors, prob=prob_anchors, valid_mask=np.ones(n_anchor, bool), axes=("trace", "time"), time_axis=1)
PathTopology(kind="segment_1d", component_id=anchor_segment_id, valid_mask=...)
trace_global_path(...)
```

### 6. restore to all gather traces

Use existing `restore_anchor_predictions_to_full_traces()` with:

```text
anchor_pick_i_raw      from projected GPT / argmax j
anchor_pmax            selected prob
trace_valid            all True for anchor positions
anchor_source_pos      anchor_positions
segment_id             anchor_segment_id
segments               full gather segments
n_traces               gather_n_traces
n_samples_orig         raw_time_len
```

Then write final gather result into global file arrays using:

```python
for gather_pos, raw_trace_idx in enumerate(raw_trace_indices):
    final arrays[raw_trace_idx] = restored[gather_pos]
```

This should replace the old tile-level weighted sum path for GPT enabled. For GPT disabled, keep existing argmax tile merge behavior unless refactoring naturally shares code.

### 7. pending accumulator validation

At end of `run_coarse_infer()`, if any accumulator has incomplete tiles, raise `RuntimeError` with gather id and missing tile ids.

### 8. QC behavior for this issue

If QC is enabled and GPT enabled with tiling, it is acceptable to either:

- skip GPT-specific QC with a warning until Issue 008, or
- write one QC per flushed gather using a representative / merged score view.

Do not write misleading tile-local GPT QC.

## テスト要件

新規推奨:

```text
packages/seisai-engine/tests/test_fbpick_coarse_gpt_score_merge.py
```

最低限:

```python
def test_gpt_score_accumulator_merges_duplicate_anchor_positions_by_weighted_probability() -> None: ...
def test_gpt_score_accumulator_renormalizes_probability_rows() -> None: ...
def test_gpt_score_accumulator_rejects_inconsistent_overlap_raw_indices() -> None: ...
def test_gpt_score_accumulator_flushes_only_after_all_tiles_seen() -> None: ...
def test_gpt_score_accumulator_builds_full_gather_segments_not_tile_segments() -> None: ...
def test_run_coarse_infer_gpt_enabled_accepts_channel_tiling_after_merge() -> None: ...
```

Synthetic tile merge test:

- gather length 12
- tile 0 span [0, 8)
- tile 1 span [4, 12)
- overlap [4, 8)
- same anchor position appears in both tiles with different probability distributions
- expected merged probability is weighted mean followed by normalization

## 受け入れ条件

- GPT enabled + channel tiling が実行できる。
- GPT は tile ごとではなく gather ごとに 1 回だけ実行される。
- gap segmentation は full gather offsets で決まる。
- existing GPT disabled behavior は維持される。
- pending / inconsistent tile overlap は明確に fail する。

## 推奨確認コマンド

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse_gpt_score_merge.py \
       packages/seisai-engine/tests/test_fbpick_coarse.py -q
```
